RFC/src/RFC-0140 Syncing and seeding.md

```diff
@@ -0,0 +1,131 @@
+
+After this process the node will be in sync and able to process blocks and transaction normally. 
+
+#### Keeping in sync
+
+The node should periodically test its peers with ping messages to ensure that they are alive. When a node sends a ping message, it should enclude the current total pow, hash of the [current head](currenthead) and genesis block hash of its own current longest chain in the ping message. The receiving node will replay with a pong message also including the total pow, [current head](currenthead) and genesis block hash of its longest chain. If the two chains dont match up, the node with the lowest pow has the responsiblity to ask the peer for syncing information. 
+
+If the genesis block hash's dont match the node is removed from its peer list as this node is running a different block chain, we can remove it. 
+
+This will be hanled by the node asking for each block from the [current head](currenthead) going backword untill a known block is found. If no block is found, the node will enter sync mode and resync. It cannot recover from its state as the fork is older than its [pruning horizon](pruninghorizon).
+
+#### Chain forks
+
+Chain forks can be a problem since in mimblewimble not all nodes keep the complete history, the design philosophy  is more in the lines of only keeping the current [utxo]. Only keepinng the current [utxo] will not be possible, because if you encounter a fork where there are two running version op the blockchain, you will not be able to swop without doing sync again. 
+
+To counter this problem we use  [pruning horizon](pruninghorizon), this allows every [node](base node) to be a "light" [archival node](archivenode). This in effect means that the node will keep a full history for a short while. If the node encounters a fork it can easily rewind its state to apply the fork. If the fork is longer than the [pruning horizon](pruninghorizon), the node will enter a sync state where it will resync. 
+
+### Pruning and cut-through
+
+[Pruning and cut-through]: #Pruning-and-cut-through	"Remove already spent outputs from the [utxo]"
+
+In MimbleWimble, the state can be completely verified using the current [UTXO](utxo) set, the set of excess signatures (contained in the transaction kernels) and the proof-of-work. The full block and transaction history is not required. This allows base layer nodes to remove old used inputs from the [blockchain] and or the [mempool]. [Cut-through](cut-through) happens in the [mempool] while pruning happens in the [blockchain] with already confirmed transactions. This will remove the inputs and outputs, but will retain the excesses  of each [transaction]. 
+
+Pruning is only for the benefit of the local base node as it reduces the local blockchain size. Pruning only happens afterthe [pruning horizon](pruninghorizon) height. A Base node will either run in archive mode or prune mode, if the base node is running in archive mode it should not prune. 
```

Comment:
I interpret "after" to mean _newer_ here. Perhaps stick to "older/newer" vs "after/before" which is less ambiguous